core/src/test/java/org/infinispan/partitionhandling/IndependentClustersMergeTest.java

```diff
@@ -0,0 +1,66 @@
+package org.infinispan.partitionhandling;
+
+import static org.testng.AssertJUnit.assertEquals;
+
+import org.infinispan.Cache;
+import org.infinispan.configuration.cache.ConfigurationBuilder;
+import org.infinispan.container.entries.CacheEntry;
+import org.infinispan.manager.EmbeddedCacheManager;
+import org.infinispan.test.TestingUtil;
+import org.infinispan.test.fwk.TransportFlags;
+import org.testng.annotations.Test;
+
+@Test(groups = "functional", testName = "partitionhandling.IndependentClustersMergeTest")
+public class IndependentClustersMergeTest extends BasePartitionHandlingTest {
+
+   private static final String MERGE_RESULT = "merge-result";
+
+   @Override
+   public Object[] factory() {
+      return new Object[] {
+            new IndependentClustersMergeTest().partitionHandling(PartitionHandling.ALLOW_READS),
+            new IndependentClustersMergeTest().partitionHandling(PartitionHandling.DENY_READ_WRITES)
+      };
+   }
+
+   public IndependentClustersMergeTest() {
+      this.numMembersInCluster = 2;
+      this.mergePolicy = (preferredEntry, otherEntries) -> {
+         CacheEntry entry = preferredEntry != null ? preferredEntry : (CacheEntry) otherEntries.get(0);
```

Comment:
Unnecessary cast. Could also avoid the unchecked warning by declaring `entry` and `mergePolicy` with generic parameters `<String, String>`.